File: src/main/java/dev/ambryn/task/utils/MathUtils.java
package dev.ambryn.task.utils;

public class MathUtils {
    public static Long factorial(int n) {
        if (n == 0 || n == 1) return 1L;
        return n * factorial(n - 1);
    }

    public static Long fib(int n) {
        if (n < 2) return (long) n;
        return fib(n - 2) + fib(n - 1);
    }
}

File: src/main/java/dev/ambryn/task/utils/MathUtils.kt
package dev.ambryn.task.utils

object MathUtils {
    fun factorial(n: Int): Long {
        return if (n == 0 || n == 1) 1L else n * factorial(n - 1)
    }

    suspend fun fib(n: Int): Long {
        return if (n < 2) n.toLong() else fib(n - 2) + fib(
            n - 1
        )
    }
}
